### ISSUE TITLE
Underlying calculations 

### ISSUE THREAD

**Issue by ahartman** [claytonjn/hass-circadian_lighting#55]
I am running your module at home and we like the outcome.
However, for production somewhere else, I have to do this in Node-RED in a situation where Home Assistant is not available.
Would it be possible for you to explain the formulas you're using?

That would help many people to understand the logic in your module and it would help me enormously towards realising the sale in Node-RED.

I took a long look in your code and as far as I can see it is all about calculating self.data['percent']. Applying that to color temperature and brightness is easy.
Hope to hear from you.
Regards, ahartman

**Comment by claytonjn** [comment]
There's a decent description in the commit here: https://github.com/claytonjn/hass-circadian_lighting/commit/4f332a514ccf491ef1bcdb171b33d36f759679a0

Ultimately though, the better solution is to figure out an equation that can calculate color temperature from only the lat/long and sun elevation, not taking into account the time of day or anything like that.

**Comment by ahartman** [comment]
Thank you, I will study that;
In fact, I am particularly interested in calculating brightness which stays at 100% a long time on both sides of noon in your module. There are modules available in Node-RED giving the percentage of the sun position in relation to noon, so I do not have to worry about the, although that does not give me a solution for before and after sunrise.

Regards, ahartman

**Comment by claytonjn** [comment]
Brightness is 100% from sunrise to sunset. My thought process there is that once the sun is fully above the horizon the amount of light doesn't really change. It only starts to dim as the sun sets below the horizon, but I haven't found any API to get the time when the sun starts setting so I just use sunset.

**Comment by ahartman** [comment]
I really like the effect but I think the system is incomplete. The sun describes a daily arc as well as a seasonal arc. Your system models the daily arc but it could maybe also do the seasonal arc. In winter the maximum color temperature is lower than in summer. I have no idea how to do that but it would make the system more complete as well as healthier as many people suffer especially in winter from light conditions.
regards, ahartman

**Comment by claytonjn** [comment]
I completely agree. Like I said, we need an equation that can calculate color temperature from only the lat/long and sun elevation - the change in max colortemp is because of a change in maximum sun elevation.

**Comment by ahartman** [comment]
I think the daily arc and seasonal arc are two different mechanisms. The daily arc will go to the maximum color temperature - 100% - every day. In my opinion the seasonal arc should limit the maximum color temperature for each day; the maximum color temperature itself should be set for the lognest day in the year, 20 or 21 June. So, 100% color temperature in the daily arc wil be a warmer white in winter than in summer.

There are quite a number of sun-position plugins in Node-RED as well as in Homebridge. Since they are all Javascript and they are all based on the same SunCalc library, it should not be difficult to see how to extend your HA component.

Renards,ahartman
